### DISCUSSION TITLE
A generic way to know if application running on TeaVM (not GWT) in core project

### DISCUSSION THREAD

**Discussion by antzGames** [xpenatan/gdx-teavm#101] [Q&A]
I need a way to determine if the application is running on teaVM **_in the core project_** other than using ```ApplicationType.WebGL```

I think this is an important question because I am sure that there are many existing libGDX projects and libraries that use ```ApplicationType.WebGL``` to point to native GWT methods.

Now with teaVM, we can use full java/kotlin implementations.

Anyhow, is there any field or property that can be access in the core project that allows me to know if the application is running in TeaVM?

Side note:  Should we ask libGDX maintainers to add ```ApplicationType.TeaVM```?  

BTW, I have a solution to this problem but it is not elegant.

**Comment by xpenatan** [comment]
You want to know if WebGL is teavm or GWT ?

I think we can use system property to check it.  I added a useragent from javascript. We can add more properties.

https://github.com/xpenatan/gdx-teavm/blob/master/examples/core/core/src/main/java/com/github/xpenatan/gdx/examples/tests/GearsDemo.java#L71

**Comment by antzGames** [comment]
So the OR is mutually exclusive, which means I need to differentiate if its running TeaVM exclusively.  Maybe an pseudo code if statement will explain it better:

```
if (teaVM) {
  ...
} else if (GWT) {
   ....
} else {
 ...
}
```
